include/neighbor.h

```diff
@@ -49,37 +50,14 @@ namespace diskann {
   };
 
   static inline unsigned InsertIntoPool(std::vector<Neighbor> &neighbors,
-                                        unsigned K, Neighbor nn) {
+                                        unsigned K, unsigned cap, Neighbor nn) {
     // find the location to insert
-    unsigned left = 0, right = K - 1;
-    if (neighbors[left].distance > nn.distance) {
-      neighbors.insert(neighbors.begin() + left, nn);
-      return left;
+    auto it = std::lower_bound(neighbors.begin(), neighbors.begin() + K, nn);
```

Comment:
In general your changes are a vast improvement. It would be good if the indexer K could be removed from the function such that this could be done:
auto it = std::lower_bound(neighbors.begin(), neighbors.end(), nn);
instead of
auto it = std::lower_bound(neighbors.begin(), neighbors.begin() + K, nn);
The current calling code isn't' structured in a way that makes this a simple change. In some cases there's an assumption that the vector is prepopulated (etc).
I'm heading out on vacation until late next week so I don't have time to dig into this. If you have a tidy way to change this that would be great otherwise I'll revisit this when I'm back. thanks -

Reply:
OK. I'll give it a try.